win32ss/user/winsrv/usersrv/harderror.c

```diff
@@ -1136,6 +1119,16 @@ UserServerHardError(
         // (Message->UnicodeStringParameterMask & 0x3)
     }
 
+    Status = NtUserSetInformationThread(NtCurrentThread(),
+                                        UserThreadUseActiveDesktop,
+                                        NULL,
+                                        0);
```

Comment:
What about doing this call *just before* calling the functions that will output GUI elements (before UserpShowInformationBalloon() / UserpMessageBox() calls)? This will allow any logging of the hard-error message (see the UserpLogHardError() call below) even if for some reason we don't have an active desktop yet to be assigned (or if another error happens that makes this call failing).